packages/suite/src/components/suite/AppRouter.tsx

```diff
@@ -1,24 +1,39 @@
 import { ComponentType, createElement, memo } from 'react';
-import { Route, Routes } from 'react-router';
 
 import { PageName } from '@suite-common/suite-types';
 
 import routes from 'src/constants/suite/routes';
+import { useSelector } from 'src/hooks/suite';
+import { selectRouteName } from 'src/reducers/suite/routerReducer';
 
 type AppRouterProps = {
     components: Record<PageName, ComponentType>;
 };
 
-export const AppRouter = memo(({ components }: AppRouterProps) => (
-    <Routes>
-        {routes
-            .filter(route => Object.keys(components).includes(route.name))
-            .map(route => (
-                <Route
-                    key={route.name}
-                    path={`${process.env.ASSET_PREFIX}${route.pattern}${route.hasNestedRoutes ? '/*' : ''}`}
-                    element={createElement(components[route.name as PageName])}
-                />
-            ))}
-    </Routes>
-));
+export const AppRouter = memo(({ components }: AppRouterProps) => {
+    const routeName = useSelector(selectRouteName);
+
+    // Resolve component by route name, with nested routes falling back to parent component
+    let componentName = routeName;
+    // NOTE: This throws a TS error becuase routeNames also contain foreground app names
+    if (routeName && !(components as Record<string, ComponentType>)[routeName]) {
+        const current = routes.find(r => r.name === routeName);
+        if (current?.isNestedRoute) {
+            const parent = routes.find(
+                r => r.hasNestedRoutes && current.pattern.startsWith(`${r.pattern}/`),
+            );
+            if (parent) componentName = parent.name;
+        }
+    }
+
+    const componentToRender =
+        componentName && (components as Record<string, ComponentType>)[componentName];
+    if (componentToRender) {
+        return createElement(componentToRender);
+    }
```

Comment:
Since we have to cast I'd rather do this, it's IMO clearer
```suggestion
    if (componentName && components.hasOwnProperty(componentName)) {
        return createElement(components[componentName] as PageName);
    }
```

Reply:
ESLINT: `Do not access Object.prototype method 'hasOwnProperty' from target object.`

Reply:
Oh FFS we have this annoying thing. Yeah this is the ESLint-compliant way:
```
Object.prototype.hasOwnProperty.call(components, componentName)
```

:thought_balloon: TBH I hate this rule, if you want to override properties of prototype you can, and the consuming code does not have to know where in the chain of inheritance it was defined, that's like the point of OOP.
This rule makes sense for external input where you want to treat the the incoming object in a certain predictable way, prevent it from breaking your code.

Reply:
This is not great to write it tehre :D

Reply:
We have it all over the codebase but OK it's your call.. I personally have no problem with checking falsiness `components[componentName]` just TS does :laughing: